api/src/main/java/com/codeforcommunity/dto/emailer/EmailAttachment.java

```diff
@@ -0,0 +1,111 @@
+package com.codeforcommunity.dto.emailer;
+
+import com.codeforcommunity.aws.EncodedImage;
+import com.codeforcommunity.dto.ApiDto;
+import com.codeforcommunity.exceptions.BadRequestImageException;
+import com.codeforcommunity.exceptions.HandledException;
+
+import org.simplejavamail.api.email.AttachmentResource;
+
+import javax.mail.util.ByteArrayDataSource;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+public class EmailAttachment extends ApiDto {
+    private String name;
+    private String data;
+
+    public EmailAttachment(String name, String data) {
+        this.name = name;
+        this.data = data;
+    }
+
+    private EmailAttachment() {}
+
+    private static String getFileExtension(String base64Image) {
+        // Expected Base64 format: "data:image/{extension};base64,{imageData}"
+
+        if (base64Image == null || base64Image.length() < 10) {
+            return null;
+        }
+
+        String[] base64ImageSplit = base64Image.split(",", 2); // Split the metadata from the image data
+        if (base64ImageSplit.length != 2) {
+            return null;
+        }
+
+        String meta = base64ImageSplit[0]; // The image metadata (e.g. "data:image/png;base64")
+        String[] metaSplit = meta.split(";", 2); // Split the metadata into data type and encoding type
+
+        if (metaSplit.length != 2 || !metaSplit[1].equals("base64")) {
+            // Ensure the encoding type is base64
+            return null;
+        }
+
+        String[] dataSplit = metaSplit[0].split(":", 2); // Split the data type
+        if (dataSplit.length != 2) {
+            return null;
+        }
+
+        //String[] data = dataSplit[1].split("/", 2); // Split the image type here (e.g. "image/png")
+        //if (data.length != 2 || !data[0].equals("image")) {
+        //    // Ensure the encoded data is an image
+        //    return null;
+        //}
```

Comment:
⛏️ Remember to clean up any blocks that are commented out! If you don't need this code, make sure to delete it